packages/storage-plugin/src/symbols.ts

```diff
@@ -59,9 +59,9 @@ export const NGXS_STORAGE_PLUGIN_OPTIONS = new InjectionToken('NGXS_STORAGE_PLUG
 export const STORAGE_ENGINE = new InjectionToken('STORAGE_ENGINE');
 
 export interface StorageEngine {
-  readonly length: number;
-  getItem(key: string): any;
-  setItem(key: string, val: any): void;
-  removeItem(key: string): void;
-  clear(): void;
+  readonly length: () => Promise<number>;
```

Comment:
this is breaking changes

Reply:
yeah it is

Reply:
I can not allow to accept it if compatibility breaks

Reply:
That sounds reasonable, still wanted to propose that change since its something minor that would make any asynchronous storage possible with minimal changes to the interface

Reply:
https://github.com/ngxs/store/pull/669#issuecomment-441026846

Add optional async engine